### PR TITLE
Set default collation to utf8mb4

### LINKED ISSUE #38085: [Issue] Set default collation to utf8mb4
This issue is automatically created based on existing pull request: magento/magento2#35174: Set default collation to utf8mb4

---------
### Description (*)
MySQL 8 interprets utf8 as utf8mb3, which is deprecated:
https://dev.mysql.com/doc/relnotes/mysql/8.0/en/news-8-0-24.html
https://dev.mysql.com/doc/refman/8.0/en/charset-unicode-utf8mb3.html
We should therefor ensure utf8mb4 is used.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [x] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by sippsolutions**
### Description (*)
MySQL 8 interprets utf8 as utf8mb3, which is deprecated:
https://dev.mysql.com/doc/relnotes/mysql/8.0/en/news-8-0-24.html
https://dev.mysql.com/doc/refman/8.0/en/charset-unicode-utf8mb3.html
We should therefor ensure utf8mb4 is used.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [x] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#38085: Set default collation to utf8mb4

### CODE REVIEW

**Comment by sippsolutions** [comment]
@magento run all tests

**Comment by Den4ik** [comment]
Hi @sippsolutions
Thanks for this PR.
Could you please fix failed tests

**Comment by sippsolutions** [comment]
@Den4ik did you see my changes? the tests are intended to fail.

**Comment by Den4ik** [comment]
Hi @sippsolutions 
Sorry for delay. I don't see any commits after my message.
Unit and Static tests failed on changed file

**Comment by sippsolutions** [comment]
@Den4ik that would be awesome :)

**Comment by Den4ik** [comment]
@sippsolutions Could you resolve merge conflict?

**Comment by hostep** [comment]
This might be a bit more complex than what it sounds like.

See https://github.com/OpenMage/magento-lts/pull/1036 where an attempt is made to apply a similar change to Magento 1, it contains a bunch of interesting information in the comments.

**Comment by Quazz** [comment]
https://github.com/magento/magento2/blob/0a834cce01c3d924d450c9a71123d9ac6de68b62/lib/internal/Magento/Framework/DB/Ddl/Table.php#L238

Would also need changes, otherwise you'll get mismatched collations which will kill your performance.

**Comment by engcom-Hotel** [comment]
@magento create issue

**Review by engcom-Hotel — CHANGES_REQUESTED** [review]
Hello @sippsolutions,

Thanks for the contribution!

In order to process further this PR please resolve the conflicts.

Meanwhile, we are moving this PR `On Hold`.

Thanks

**Comment by engcom-Hotel** [comment]
Hello @sippsolutions,

Have you got chance to check this [comment](https://github.com/magento/magento2/pull/35174#pullrequestreview-1679245011)?

Thanks

**Comment by sippsolutions** [comment]
@engcom-Hotel done

**Comment by engcom-Hotel** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Review by engcom-Hotel — CHANGES_REQUESTED** [review]
Hello @sippsolutions,

Thanks for the contribution!

Please fix the integration tests failures. It seems they are due to the PR changes.

Thanks

**Comment by engcom-Hotel** [comment]
@sippsolutions we are picking up this PR for fixes.

**Comment by engcom-Hotel** [comment]
@magento run Integration Tests

**Comment by engcom-Hotel** [comment]
@magento run Integration Tests

**Comment by engcom-Hotel** [comment]
@magento run Integration Tests

**Comment by engcom-Hotel** [comment]
@magento run Integration Tests

**Comment by engcom-Hotel** [comment]
@magento run Integration Tests

**Comment by engcom-Hotel** [comment]
@magento run Integration Tests

**Comment by engcom-Hotel** [comment]
@magento run Integration Tests

**Comment by engcom-Hotel** [comment]
@magento run Integration Tests

**Comment by engcom-Hotel** [comment]
@magento run Integration Tests

**Comment by engcom-Hotel** [comment]
@magento run Database Compare, Functional Tests B2B, Functional Tests CE, Functional Tests EE, Magento Health Index, Semantic Version Checker, Static Tests, Unit Tests, WebAPI Tests

**Comment by engcom-Hotel** [comment]
@magento run Static Tests

**Comment by engcom-Hotel** [comment]
@magento run Database Compare, Functional Tests B2B, Functional Tests CE, Functional Tests EE, Integration Tests, Magento Health Index, Semantic Version Checker, Unit Tests, WebAPI Tests

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B, Functional Tests CE, Functional Tests EE, WebAPI Tests

**Comment by engcom-Hotel** [comment]
@magento run WebAPI Tests

**Comment by engcom-Hotel** [comment]
The failed tests seem falky to me. Please refer the below screenshot for reference:

**Functional Tests CE**
image

image

**Functional Tests B2b**
image

image

**Functional Tests EE**
image

image

For the `Database Compare` test failure we need to raise an approval ticket. Hence moving this PR further.

Thanks

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B, Functional Tests CE, Functional Tests EE

**Comment by Den4ik** [comment]
@magento run Functional Tests CE, Functional Tests EE

**Comment by engcom-Hotel** [comment]
Hello @Den4ik,

Are you still working on this PR for Database compare test failures?

Moving this PR to the changes requested.

Thanks

**Comment by andrewbess** [comment]
Hello @Den4ik @engcom-Hotel 

The `Database Compare` tests work good.
It compares database structure before fixes and after it.
So, the DB structure will be different after these fixes.

@engcom-Hotel please create approval ticket in the internal Jira to move it further.

Thank you in advance.

**Comment by andrewbess** [comment]
Hello @sippsolutions 
Thank you for your collaboration.
Changes look good for me ✔️.
Let's wait for passing functional tests.

**Review by Den4ik — APPROVED** [review]
Functional tests passed in previous runs.
Checked DB tests and it looks ok

**Review by andrewbess — APPROVED** [review]
Approved

**Comment by andrewbess** [comment]
@magento run all tests

**Comment by engcom-Delta** [comment]
@magento run all tests

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests CE, Functional Tests B2B

**Comment by engcom-Bravo** [comment]
Hi,

Internal team has started working on this issue. Most of the code changes are complete. We are trying to get conclusion over DBC failure and are in discussion with PO. 

Thanks.

**Comment by engcom-Hotel** [comment]
As per this comment https://github.com/magento/magento2/pull/35174#issuecomment-1916483549, we are moving this issue `On hold`.

We will update you here about the progress.

Thanks

**Comment by engcom-Bravo** [comment]
Hi,

All the required changes are done and working with Engineering Manager to speed up. Delivery is aligned for Oct Release.

Thanks.

**Comment by hostep** [comment]
@engcom-Bravo: what would be the status here? Is everything still in order for the October release? (I guess that means Magento 2.4.8-beta1?)

**Comment by engcom-Bravo** [comment]
Hi @hostep @sippsolutions,

Unfortunately this did not make it to 2.4.8-beta1 release. The implementation is close to completion now and will be followed by extensive testing based on its large scope. We will keep you updated on the progress. Appreciate your patience!

Thanks.

File: dev/tests/setup-integration/_files/Magento/TestSetupDeclarationModule1/revisions/old_diff/InstallSchema.php
Comment (glo71317) [inline dev/tests/setup-integration/_files/Magento/TestSetupDeclarationModule1/revisions/old_diff/InstallSchema.php:6]: Why we are ignoring file, try to resolve this

File: dev/tests/setup-integration/_files/Magento/TestSetupDeclarationModule1/revisions/base_update/InstallSchema.php
Comment (glo71317) [inline dev/tests/setup-integration/_files/Magento/TestSetupDeclarationModule1/revisions/base_update/InstallSchema.php:8]: Why we are ignoring file, try to resolve this

**Review by glo71317 — CHANGES_REQUESTED** [review]
Need to perform the upgrade with pr changes from Adobe commerce lower version

**Comment by glo71317** [comment]
> Need to perform the upgrade with pr changes from Adobe commerce lower version

Please ignore my comment above. This was intended for internal team. Apologies for any confusion that it may have caused.

**Pull request merged**
